publisher_connection/connection.go

```diff
@@ -0,0 +1,16 @@
+package publisher_connection
+
+type ConnectionDetails struct {
```

Comment:
actually there is no schema for this part of the json.. it's depends on the publisher. 
so maybe it's better to make it interface{}